From 879fc3da5b7df9625db674a0089f775b76bd3abf Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:21 -0400
Subject: [infiniband] ocrdma: use right macro in query ah

Message-id: <7b20dc83519c49f50d93c8ef9130b8291cc2dc95.1411571481.git.dledford@redhat.com>
Patchwork-id: 96634
O-Subject: [Patch RHEL7 46/48] RDMA/ocrdma: use right macro in query ah
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

query-ah does not uses correct macro and checks the wrong bit
for the validity of address handle in vector table.
This patch is a fix for it.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
(cherry picked from commit 28386adcdf75d572d1d02c0a9bddf3d6b725822f)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_ah.c b/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
index 8d21e33..ab29346 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
@@ -164,7 +164,7 @@ int ocrdma_query_ah(struct ib_ah *ibah, struct ib_ah_attr *attr)
  struct ocrdma_av *av = ah->av;
  struct ocrdma_grh *grh;
  attr->ah_flags |= IB_AH_GRH;
- if (ah->av->valid & Bit(1)) {
+ if (ah->av->valid & OCRDMA_AV_VALID) {
   grh = (struct ocrdma_grh *)((u8 *)ah->av +
     sizeof(struct ocrdma_eth_vlan));
   attr->sl = be16_to_cpu(av->eth_hdr.vlan_tag) >> 13;
-- 
1.7.1